src/NBomber/Plugins/PsPingPlugin.fs

```diff
@@ -10,27 +10,37 @@ open FSharp.Control.Tasks.NonAffine
 open FsToolkit.ErrorHandling
 open Microsoft.Extensions.Configuration
 
+open NBomber
 open NBomber.Contracts
+open NBomber.Domain.Stats.Statistics
 open NBomber.Extensions.InternalExtensions
 
 [<CLIMutable>]
 type PsPingPluginConfig = {
     Hosts: Uri[]
     /// The default is 1000 ms.
     Timeout: int
+    /// Number of ping executions. The default is 4.
+    Executions: int
 } with
-    static member CreateDefault([<ParamArray>]hosts: string[]) = {
+    static member CreateDefault([<ParamArray>]hosts: string[]) =
+        {
             Hosts = hosts |> Array.map Uri
             Timeout = 1_000
-    }
+            Executions = 4
+        }
 
     static member CreateDefault(hosts: string seq) =
         hosts |> Seq.toArray |> PsPingPluginConfig.CreateDefault
 
 type PsPingReply = {
-    Status: string
     Address: Uri
-    RoundtripTime: int64
+    RequestCountSucceeded: int
+    RequestCountFailed: int
+    RoundtripTimeMin: int64
+    RoundtripTimeMax: int64
+    RoundtripTimeAvg: float
```

Comment:
Let's simplify it :)
Let's have only RoundtripTime where you will use avg time

Reply:
Well, I need all these, from home office I get something very crazy roundtrip times, and want to know about them, e.g. 2  runs done with 15 minutes difference, both from home (not sure what is happening):

![image](https://user-images.githubusercontent.com/25847560/142471503-44bf2043-24f4-40b5-9f41-f06dad3db1df.png)

![image](https://user-images.githubusercontent.com/25847560/142471554-1d85263c-923c-4f08-9434-9cf00462f80a.png)

Reply:
yes, but do u understand that if you have > 2 ms you can stop testing

Reply:
and I don't see a big pint to have min, max here, and ok, fail
it's PING not something critical

Reply:
Well, I saw the warning you have built in, and was wondering why so. My serverside request duration is between 70 and 100ms in the best case and depending on the load goes up to 300ms per request. Why would I be bothered by 50ms pings? As long as they are stable, I can safely subtract 50ms from the results of my executions, and I am fine ... e.g.:

![image](https://user-images.githubusercontent.com/25847560/142474030-4406e234-44ee-416b-a00e-4f06c61f348c.png)

Reply:
here another example with 10 pings: 
![image](https://user-images.githubusercontent.com/25847560/142474254-99794fd4-6036-43bf-b448-96f45e52c4e4.png)

Reply:
When you do load testing you want to know how much time your server takes to process requests.
And for correct results, you shouldn't count any physical latency, meaning that you want to test your target system in isolation.
That's why 2ms
< 2ms means that you test your system in an almost isolated env, and the figures that NBombers show will show you the truth.

Reply:
Any additional physical latency will bring inaccuracy in your results, and you will not be able to subtract PING from your processing time correctly.

Reply:
that's the point of 2ms

Reply:
Host | Status | Address | Round Trip Time | Time to Live | Don't Fragment | Buffer Size
-- | -- | -- | -- | -- | -- | --
nbomber.com | Success | 104.248.140.128 | 43 ms | 128 | False | 32 bytes

<ol style="box-sizing: border-box; margin: 0 0 var(--ifm-list-margin); padding-left: var(--ifm-list-left-padding); color: rgb(245, 246, 247); font-family: system-ui, -apple-system, &quot;Segoe UI&quot;, Roboto, Ubuntu, Cantarell, &quot;Noto Sans&quot;, sans-serif, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(24, 25, 26); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li style="box-sizing: border-box; overflow-wrap: break-word;"><p style="box-sizing: border-box; margin-top: var(--ifm-list-paragraph-margin); margin-right: ; margin-bottom: ; margin-left: ;">The first noticeable thing is physical latency (<strong style="box-sizing: border-box; font-weight: var(--ifm-font-weight-bold);">Ping statistics -&gt; Round Trip Time: 43ms</strong>) between the load test agent and the target host. It shows that you run your test under problematic infrastructure (internet connection is bad or your test agent and target system located in different datacenters). You should run your tests where are physical latency will be close to 0 - 1ms.</p></li></ol>

Reply:
@deyanp read this
https://nbomber.com/docs/loadtesting-basics#what-to-measure-about-performance

Reply:
@AntyaDev, I have read this, and understand that the best thing is to run the tests from within the cloud infrastucture where also the service is located (I am on my way there, it just take a bit more time to properly configure the fragile Azure DevOps pipelines ...). But in practical terms, I (and probably other people too) take shortcuts, one of them being "let me quickly run a couple of load tests from my i9 16 core machine which is 10x more powerful than any VM I will provision in the cloud (or the sum of *all* VMs I have provisioned in the cloud ;))). And for that purpose the physical ping becomes interesting, and usually it is pretty stable (e.g. from my home network I have 60-70ms to Azure and GCP West Europe). Subtracting is then a no-brainer, and a couple of milliseconds inaccuracy are not at all important in my case.

Reply:
+ what if you want to load tests 2 setups in 2 different clouds (e.g. Azure and GCP) from a central location, instead of installing the load test runners in each?

Reply:
> Subtracting is then a no-brainer, and a couple of milliseconds inaccuracy are not at all important in my case.

That's what I mean, for such a case why complicate the solution?

Reply:
It's already clear that you want to run kind of dirty test to see that your system handles 10K and not 100
Yes, I understand this.
But why then we should introduce all these parameters to configure your PING logic for such a hacky case?
My problem is that we bring functionality for the non-production use cases.

Reply:
What % of NBomber users do you thing use it for "clean" cases and what % for "hacky" cases?
Besides, I don't think adding min/mean/max/stdev and Executions configurability is doing any harm whatsoever, default for Executions will be 3 or 4 anyway, and the details can be ignored by those who do not need them. Not sure I see the problem, but again, this is your project, it is up to you to decide what you want to have and what not ..

Reply:
Can you at lest make it possible for having External Pluguns, not part of the NBomber Solution? Currently this seems to be impossibke, due to a dependency on an internal module ...

Reply:
I meant NBomber.​Extensions​.​InternalExtensions

Reply:
not sure that I understand the problem.

Reply:
I cannot currently extract the PsPingPlugin and have outside of the NBomber project due to this dependency ... if I could then I would have my version of PsPingPlugin as I need it, and you can have the cut-down version without min/max/stdev and Executions config.

Reply:
sure, I will take later on it.
if you want send PR